action.yml

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SSH to Google Cloud Platform compute instances
+author: Google LLC
+description: |-
+  Use this action to connect to Google Cloud Platform instances via ssh and execute user’s commands.
+
+inputs:
+  instance_name:
+    description: |-
+      Name of the virtual machine instance to SSH into.
+    required: true
+
+  zone:
+    description: |-
+      Zone of the instance to connect to.
+    required: true
+
+  user:
+    description: |-
+      Username with which to SSH. If omitted, the user login name is used. 
+      If using OS Login, USER will be replaced by the OS Login user.
+    required: false
+
+  container:
+    description: |-
+      The name or ID of a container inside of the virtual machine instance to connect to.
+      This only applies to virtual machines that are using a Google Container-Optimized 
+      virtual machine image.
+    required: false
+
+  ssh_key_file_path:
+    description: |-
+      Absolute path to the SSH key file. By default, this is '/home/runner/.ssh/google_compute_engine'.
+    required: false
```

Comment:
Maybe I am missing something but if a user does not own the runner it seems suboptimal to expect them to have another pre step to create this file. Is there a reason why we cant get the contents  (user will provide via GH secret) and write this to a file ourselves?

Reply:
I was also thinking about using GH secret, but there is one blocker that I see: there can be a situation where user may want to connect to different VM's with different keys. In this case we need somehow use different GH secrets. WDYT?

Could you please describe a situation, where user doesn't have access for runner files, but our action has in the same time?

Reply:
Wouldn't it be better to not use SSH keys at all and instead rely on IAP and os-login?

Reply:
@sethvargo @bharathkkb

But what if VM doesn't have os-login? Not 100% sure, but I would prefer to let users use their ssh keys. If you want to use ssh keys, you have to prepare them before using action (like in the example workflow).

There are several questions here:
1) Security. Not sure if we have any risks by storing keys on the runner filesystem.
2) Do we want to generate ssh keys if user doesn't provide them? In this case we will update project metadata also.
3) As a variant for the first version we can rely only on IAP and os-login, but not sure.